pkg/metrics/tsp_metric_collector.go

```diff
@@ -18,23 +17,38 @@ import (
 
 type TspMetricCollector struct {
 	client.Client
-	resourceCpuMetric *prometheus.Desc
-	resourceMemMetric *prometheus.Desc
+	resourceMetric *prometheus.Desc
+	externalMetric *prometheus.Desc
+	resourceMetricByWindow *prometheus.Desc
+	externalMetricByWindow *prometheus.Desc
 }
 
 func NewTspMetricCollector(client client.Client) *TspMetricCollector {
 	return &TspMetricCollector{
 		Client: client,
-		resourceCpuMetric: prometheus.NewDesc(
```

Comment:
Why add a metric `xxxxByWindow` ?

Reply:
![image](https://user-images.githubusercontent.com/41617636/180138829-8c6ddac8-4207-4dc9-b0fc-01a2fa210116.png)
![image](https://user-images.githubusercontent.com/41617636/180139370-c7230bbf-a7b3-4926-b2b9-179de4e93ef1.png)

For example，the replicas of deployment scaled by tsp's metrics with PredictionWindowSeconds
By this metric, the event of scaleDown or scaleUp can be effectively reflected